docs/extensions/param_police.py

```diff
@@ -36,8 +36,9 @@ def show_param_warnings(app, exception):
             file_name,
             line,
         )
-    if param_warnings:
-        raise RuntimeError('Encountered text parameter type. Use annotations.')
+    # TODO: Figure out with @flying-sheep what to do about these
+    # if param_warnings:
+    #     raise RuntimeError('Encountered text parameter type. Use annotations.')
```

Comment:
Hi, I’m currently on holidays, but I’m checking out mentions, thanks for notifying me here.

Every deactivation of an automation is a step backward in maintainability. 
If one needs to, of course we can add a case-by-case deactivation of with some mechanism, but never flat out like this.

In this case, unless I’m missing something, the obvious solution is to add the parameters, then do

```py
kwargs = dict(**kwargs, chunked=chunked, chunk_size=chunk_size, layer=layer, obsm=obsm)
```

If we need this more often, we should build some decorator that fetches the annotation data from some other function or so.

Reply:
[Here's what the docs for the function look like in this PR](https://icb-scanpy--1173.com.readthedocs.build/en/1173/api/scanpy.pp.log1p.html#scanpy.pp.log1p)  (readthedocs PR builds are already useful!)

The issue is that the most generic method (i.e. the fallback) for a function is the one that get's the doc-string. There can be other methods which take more arguments that we would want to document. The question is how do we document the arguments, and how do we get tab completion?

Reply:
I've just added the arguments to the function declaration and throw an error if they are passed. It would be nice to set default values and tab completion with a bit more granularity, but this PR does not need to wait on that.